python/setup.py

```diff
@@ -84,6 +85,23 @@ def strtobool(val):
         raise ValueError("invalid truth value %r" % (val,))
 
 
+MSG_DEPR_SETUP_BUILD_FLAGS = """
+  !!
+
+        ***********************************************************************
+        The '{}' flag is being passed to setup.py, but this is
+        deprecated.
+
+        In general, if a certain component is available in Arrow C++, it will
+        automatically be enabled for the PyArrow build as well. If you want
```

Comment:
Do we know if this is not the case sometimes? The `In general` sounds confusing and we might want to just say:
```suggestion
        If a certain component is available in Arrow C++, it will
        automatically be enabled for the PyArrow build as well. If you want
```

Reply:
Ah no, I think this is always the case. The "in general" was meant to convey "in general you should just rely on that fact, but if for whatever reason you want to overrule this there is also the env variable". But I see that the placement was confusing then :)